### ISSUE TITLE
I want to train and object detect quickly.

### ISSUE THREAD

**Issue by Heidisnaps** [yhenon/keras-frcnn#110]
Hi everyone,

I want to train and object detect quickly. Even if the accuracy is a little lower.
So I want to reduce the structure of net (vgg or resnet), but I do not know where to fix it.
How can I change the structure of net?
Thanks.

**Comment by ftiguidou** [comment]
Hello,

If you want to know where are the net config files, everything is stored in resnet.py in the keras_frcnn folder. You have the nn_base function which is the common part between the RPN and the classifier and classifier_layers and rpn functions for the ending part of each.

Concerning the train process, I believe it is printed from keras.utils.generic_utils module in the lines : 
```
	progbar = generic_utils.Progbar(epoch_length)

progbar.update(iter_num, [('rpn_cls', np.mean(losses[:iter_num, 0])), ('rpn_regr', p.mean(losses[:iter_num, 1])), ('detector_cls', np.mean(losses[:iter_num, 2])), ('detector_regr', np.mean(losses[:iter_num, 3]))])
```
Hope it answered your questions !

**Comment by thealchemist89** [comment]
Hi Heidisnaps and ftiguidou,

Could you share what annotation tools (if any) you were using for your custom datasets and how the simple_parser.py could be used in getting the ROI?

Thank you so much for the help.

**Comment by ftiguidou** [comment]
Hi thealchemist89,

I did not use any annotation tool as my dataset was already labeled, sorry ^^'. The simple_parser is not directly used for getting the ROI, its goal is only to get the data and to parse it. If you want to know how is used the simple_parser, you have to write a txt file in which each line correspond to one of your pictures according to this format : filepath,x1,y1,x2,y2,class_name
This is explained in the README.

For getting the ROI, I suggest you to have a look at the roi_helpers.py. This scripts is used for calculating the IoU and turning the area proposed by the RPN net to ROI according to config parameters.

Hope it helped !

**Comment by thealchemist89** [comment]
Hi ftiguidou,

Thanks for your reply! I was thinking of using a image annotation tool that outputs in a format that could be tweaked to follow: filepath, x1,y1,x2,y2,class_name.

I will follow up from your suggestions! May I also ask, how do you train your customized dataset on-top of the pre-train model(ResNet/VGG)? Which part of the code allows us to remove the top layer and inject our dataset in?

Thanks for your time and help again!

**Comment by Heidisnaps** [comment]
Hi. @ftiguidou 
Thanks for your answer.

I tried to change the code in resnet.py. Especially, I tried to change the 'nn_base' code.
I put a '#' in front of the code to reduce the convolution layer.
But it was an error.
How can I reduce the number of convolution layers without error?
Can I simply reduce the number of convolution layers by modifying the resnet.py code?
`
def nn_base(input_tensor=None, trainable=False):

    # Determine proper input shape
    if K.image_dim_ordering() == 'th':
        input_shape = (3, None, None)
    else:
        input_shape = (None, None, 3)

    if input_tensor is None:
        img_input = Input(shape=input_shape)
    else:
        if not K.is_keras_tensor(input_tensor):
            img_input = Input(tensor=input_tensor, shape=input_shape)
        else:
            img_input = input_tensor

    if K.image_dim_ordering() == 'tf':
        bn_axis = 3
    else:
        bn_axis = 1

    x = ZeroPadding2D((3, 3))(img_input)

    x = Convolution2D(64, (7, 7), strides=(2, 2), name='conv1', trainable = trainable)(x)
    x = FixedBatchNormalization(axis=bn_axis, name='bn_conv1')(x)
    x = Activation('relu')(x)
    x = MaxPooling2D((3, 3), strides=(2, 2))(x)

    x = conv_block(x, 3, [64, 64, 256], stage=2, block='a', strides=(1, 1), trainable = trainable)
    x = identity_block(x, 3, [64, 64, 256], stage=2, block='b', trainable = trainable)
    x = identity_block(x, 3, [64, 64, 256], stage=2, block='c', trainable = trainable)


    x = conv_block(x, 3, [128, 128, 512], stage=3, block='a', trainable = trainable)
    x = identity_block(x, 3, [128, 128, 512], stage=3, block='b', trainable = trainable)
    x = identity_block(x, 3, [128, 128, 512], stage=3, block='c', trainable = trainable)
    x = identity_block(x, 3, [128, 128, 512], stage=3, block='d', trainable = trainable)

    **#x = conv_block(x, 3, [256, 256, 1024], stage=4, block='a', trainable = trainable)
    #x = identity_block(x, 3, [256, 256, 1024], stage=4, block='b', trainable = trainable)
    #x = identity_block(x, 3, [256, 256, 1024], stage=4, block='c', trainable = trainable)
    #x = identity_block(x, 3, [256, 256, 1024], stage=4, block='d', trainable = trainable)
    #x = identity_block(x, 3, [256, 256, 1024], stage=4, block='e', trainable = trainable)
    #x = identity_block(x, 3, [256, 256, 1024], stage=4, block='f', trainable = trainable)**

    return x`

**Comment by ftiguidou** [comment]
You cannot just change the network by removing layers. In your case, you are changing the output shape of the nn_base network, if you do that, you need to adapt the other networks (coming after the nn_base) that use this output as an input.

**Comment by Heidisnaps** [comment]
@ftiguidou thanks for answer.
